django_stomp/execution.py

```diff
@@ -24,6 +24,7 @@
 durable_topic_subscription = eval_str_as_boolean(getattr(settings, "STOMP_DURABLE_TOPIC_SUBSCRIPTION", False))
 listener_client_id = getattr(settings, "STOMP_LISTENER_CLIENT_ID", None)
 is_correlation_id_required = eval_str_as_boolean(getattr(settings, "STOMP_CORRELATION_ID_REQUIRED", True))
+should_process_msg_on_background = eval_str_as_boolean(getattr(settings, "STOMP_PROCESS_MSG_ON_BACKGROUND", False))
```

Comment:
I believe `STOMP_PROCESS_MSG_ON_BACKGROUND` should be enabled by default. Starting from this feature I think `STOMP_OUTGOING_HEARTBIT` and `STOMP_INCOMING_HEARTBIT` both would have a default value as well to be used as we do not have the issue you related [here](https://github.com/juntossomosmais/django-stomp/commit/09530fa9efe580337eae798cf96abb24e2897eff#diff-04c6e90faac2675aa89e2176d2eec7d8R84).

Reply:
I'm not acquainted this project's code, but here's my input: 

I don't know if this is a good idea @willianantunes. According to the [setup](https://github.com/juntossomosmais/django-stomp/pull/21/files#diff-2eeaed663bd0d25b7e608891384b7298R13), we're making a minor change (2.1.1 to 2.2.1), which implies that it won't break in any project that updates from 2.1.1 to 2.2.1. 

We made something like this in the past (setting a default value for a new feature as true) and we had some problems. I don't remember exactly what was the matter, but it was something related to Heartbit.

I think it's safer let the default value as False, and after some happy results in the production environment we can set the default value as True. What do you think?

Reply:
It would be a major change in the way `django-stomp` consumers work, if we enable this feature by default and the hearbeats.

I don't mind much, but why makes this change right now? Why not run it with this disabled by default, maintaining backward compatibility and, if no issues appear, we enable it by default?

Reply:
> According to the [setup](https://github.com/juntossomosmais/django-stomp/pull/21/files#diff-2eeaed663bd0d25b7e608891384b7298R13), we're making a minor change (2.1.1 to 2.2.1), which implies that it won't break in any project that updates from 2.1.1 to 2.2.1.

I should change it to `2.2.0`!

Reply:
@lgigek that's exactly I would like it to be. I talk to @ricardochaves to give his thoughts on this well. @muriloamendola feel free to do it too!

Reply:
> It would be a major change in the way django-stomp consumers work, if we enable this feature by default and the hearbeats.

I agree with it. Since this feature is a minor change, I like the idea of maintain compatibility of previous versions.


> I don't mind much, but why makes this change right now? Why not run it with this disabled by default, maintaining backward compatibility and, if no issues appear, we enable it by default?

I like this approach, I think it's safer proceeding this way.

Reply:
When not using heartbeat, this setting is irrelevant.
When you are going to use heartbeat, you need it enabled under penalty of having serious problems. Today we are turning off heartbeat because we understand that it is extremely dangerous to use this with a single line.
Given the scenarios above, I understand that it must be default `True`, change the version to `3.0.0` and give a log `warning` about the danger of using hearbeat with this False setting.

Reply:
After all, we all agree that if it's turned on by default, we should release a new major version. And, the only reason to not do it is if we're not sure that this feature will resolve the issue that we found when we enabled the heartbeats.

Given that, I agree with @willianantunes. This and the STOMP heartbeats should be enabled by default. Commit [55afde2](https://github.com/juntossomosmais/django-stomp/pull/21/commits/55afde2e19b9ea1dcf1cc146995989bca59ae7ec) deals with it.